API/Common.pm

```diff
@@ -107,13 +107,25 @@ sub _precacheAlbum {
 		$album->{genre} = $album->{genre}->{name};
 		$album->{image} = __PACKAGE__->getImageFromImagesHash($album->{image}) || '';
 
+		# If the user pref is for classical music enhancements to the display, is this a classical release or has the user added the genre to their custom classical list?
+		$isClassique = 0;
+		if ( $prefs->get('useClassicalEnhancements') ) {
+			my %genreList = map { $_ => 1 } split '\s*,\s*', $prefs->get('classicalGenres');	
```

Comment:
You could do this globally in the module, and update whenever there's a change to the pref. Something like:

```perl
# somewhere at the top:
my %genreList;
initGenreMap();

sub initGenreMap {
   %genreList = map { $_ => 1 } split '\s*,\s*', $prefs->get('classicalGenres');
}

$prefs->setChange(\&initGenreMap, 'classicalGenres');
```

This way you don't have to re-build the map whenever you have to pre-cache an album.

Reply:
Done.